'tach test' should return exit code zero when no tests run

Right now, if `tach test` determines that all tests can be skipped, the exit code from pytest is 5. This might mean that we should adjust the way the plugin operates (there may be a more native way to indicate tests should be skipped), adjust pytest's configuration, or maybe we need to diverge from pytest's behavior here and manually return zero in this case.

Example:
```
.venv/bin/tach test
============ Cached results found!  ============
==================================================== test session starts =====================================================
platform linux -- Python 3.12.3, pytest-8.2.2, pluggy-1.5.0
rootdir: /home/evan/Programming/tach
configfile: pyproject.toml
plugins: mock-3.14.0
collected 85 items                                                                                                           
[Tach] Skipped 10 test files (85 tests) since they were unaffected by current changes.
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_affected_modules.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_parsing.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_check.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_file_tree.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_module_tree.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_cli.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_changed_files.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_report.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_imports.py'
[Tach] > Skipped '/home/evan/Programming/tach/python/tests/test_cache.py'

=================================================== no tests ran in 0.28s ====================================================
============ END Cached results  ============
make: *** [Makefile:41: test] Error 5
```

--- a/python/tach/test.py
+++ b/python/tach/test.py
@@ -218,6 +218,13 @@ def run_affected_tests(
     )
 
     exit_code = pytest.main(pytest_args, plugins=[pytest_plugin])
+
+    if exit_code == pytest.ExitCode.NO_TESTS_COLLECTED:
+        # Selective testing means running zero tests will happen regularly,
+        # so we do not want the default behavior of failing when no tests
+        # are collected.
+        exit_code = pytest.ExitCode.OK
+
     return AffectedTestsResult(
         exit_code=exit_code,
         tests_ran_to_completion=pytest_plugin.tests_ran_to_completion,

